src/msg_recorder/impl/utils.cc

```diff
@@ -0,0 +1,18 @@
+#include "utils.h"
+
+#include <algorithm>
+#include <cctype>
+#include <string>
+
+namespace cris::core::impl {
+
+std::string GetMessageRecordFileName(const std::string& message_type, const CRMessageBase::channel_subid_t subid) {
+    auto message_record_filename   = message_type;
+    auto message_type_replace_pred = [](char c) {
+        return !std::isalpha(c) && !std::isdigit(c);
+    };
+    std::replace_if(message_record_filename.begin(), message_record_filename.end(), message_type_replace_pred, '_');
```

Comment:
Why? Some punct, e.g. `-` and `()` are totally OK in path.

Reply:
I don't want to deal with the special cases of the characters, alnums should be readable enough. Or maybe we can also allow `:` and`<>`, but nothing more.

Reply:
:and<>? They're special under windows and linux.....

Reply:
Then alnum should be fine. There should not be any other characters because it is the class name.

Reply:
`_` is ok because we convert every non-alnum chars to `_` so it will stay. Other chars will not appear in the class name.